msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/MFAStates.kt

```diff
@@ -0,0 +1,627 @@
+                    controller = NativeAuthMsalController(),
+                    publicApiId = PublicApiId.NATIVE_AUTH_MFA_DEFAULT_CHALLENGE
+                )
+
+                val rawCommandResult =
+                    CommandDispatcher.submitSilentReturningFuture(command)
+                        .get()
+
+                return@withContext when (val result =
+                    rawCommandResult.checkAndWrapCommandResultType<MFAChallengeCommandResult>()) {
+                    is MFACommandResult.VerificationRequired -> {
+                        MFARequiredResult.VerificationRequired(
+                            nextState = MFARequiredState(
+                                continuationToken = result.continuationToken,
+                                correlationId = result.correlationId,
+                                scopes = scopes,
+                                config = config
+                            ),
+                            codeLength = result.codeLength,
+                            sentTo = result.challengeTargetLabel,
+                            channel = result.challengeChannel
+                        )
+                    }
+                    is MFACommandResult.SelectionRequired -> {
+                        MFARequiredResult.SelectionRequired(
+                            nextState = MFARequiredState(
+                                continuationToken = result.continuationToken,
+                                correlationId = result.correlationId,
+                                scopes = scopes,
+                                config = config
+                            ),
+                            authMethods = result.authMethods.toListOfAuthMethods()
+                        )
+                    }
+                    is INativeAuthCommandResult.APIError -> {
+                        Logger.warnWithObject(
+                            TAG,
+                            result.correlationId,
+                            "requestChallenge() received unexpected result: ",
+                            result
+                        )
+                        MFAError(
+                            errorMessage = result.errorDescription,
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            errorCodes = result.errorCodes,
+                            exception = result.exception
+                        )
+                    }
+                    is INativeAuthCommandResult.Redirect -> {
+                        MFAError(
+                            errorType = ErrorTypes.BROWSER_REQUIRED,
+                            error = result.error,
+                            errorMessage = result.errorDescription,
+                            correlationId = result.correlationId
+                        )
+                    }
+                }
+            } catch (e: Exception) {
+                MFAError(
```

Comment:
Don't use generic MFAError?

Reply:
Take 'is INativeAuthCommandResult.Redirect -> { SubmitCodeError' as an example. Looks like we don't have separate error for BrowserRequired. Maybe MFAError?

Reply:
@Yuki-YuXin what I meant is: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2185